CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,184 @@
+---
+cip: <to be assigned>
+title: Granda Mento
+author: <a list of the author's or authors' name(s) and/or username(s), or name(s) and email(s), e.g. (use with the parentheses or triangular brackets): FirstName LastName (@GitHubUsername), FirstName LastName <foo@bar.com>, FirstName (@GitHubUsername) and GitHubUsername (@GitHubUsername)>
+discussions-to: <URL>
+status: Draft
+type: <Standards Track, Meta or Informational>
+category (*only required for Standards Track): <Ring 0, 1, 2, 3>
+created: <date created on, in ISO 8601 (yyyy-mm-dd) format>
+requires (*optional): <CIP number(s)>
+replaces (*optional): <CIP number(s)>
+license: Apache 2.0
+---
+
+## Simple Summary
+
+Granda Mento is a mechanism to facilitate large CELO <-> stable token exchanges that aren't suitable via Mento or OTC. A new contract is created that has the authorization by Governance to exchange `CELO <-> stable token`. Exchanges via this contract must be approved by a multisig and can be vetoed by Governance.
+
+## Abstract
+
+There are no existing solutions for making large exchanges ($1m+) involving stable tokens. Apart from cUSD minted for validator payments every epoch, stable tokens can only be minted via Mento. Large volume exchanges via Mento experience high slippage due to the limited bucket sizes in the constant product market maker. Granda Mento is a proposed a mechanism for large-volume stable token exchanges.
+
+## Motivation
+
+Users looking to make larger exchanges involving stable tokens are faced with 3 options:
+1. Exchange `CELO <-> stable token` via Mento slowly over a longer period of time.
+2. Arrange one or many `XXX <-> stable token` OTC trades.
+3. Create a large limit order of `fiat <-> stable token` on a centralized exchange in hopes it gets filled over time.
+
+The existing implementation of Mento is suitable for facilitating low volume exchanges, e.g. up to thousands of stable tokens at a time, with minimal slippage. However, significant slippage of over 2% starts to occur on both Mento and on centralized exchanges at trades sizes of ~$50k+. OTC trading satisfies the needs of medium volume exchanges, eg ~100k for cUSD and likely much less for stable tokens with lower total supply (like cEUR), but similarly takes a hit of around 2-3%. There are no existing avenues that are able to satisfy high volume stable token exchanges on the order of millions.
```

Comment:
It's worth calling out that options 2 and 3 above may not be possible with smaller stablecoins like cEUR and future coins with lower liquidity close to initialization. Option 3 can be done in theory, but rarely fills due to market makers capturing the spread and has too much time uncertainty to full fill.